docs/pages/tutorials/embedded-mode-with-sqlite-nextjs.mdx

```diff
@@ -0,0 +1,277 @@
+
+The tutorial will show you how to:
+
+- Setup a Next.js app
+- Embed Keystone inside it, and run an Admin UI you can read and write to locally
+- Create a simple Keystone schema with a `Post` [list](/apis/schema) that Next.js can build pages with
+- Setup a secure read-only GraphQL API endpoint (and playground) that you can access in production
+- Deploy your CMS and frontend monorepo to Vercel 🚀
+
+?> If you need access to Keystone’s Admin UI over the internet, or want to provide access to more than one editor – use Keystone in [standalone](/concepts/modes#standalone-mode) mode.
+
+## Setup a Next.js app
+
+Start out by adding a basic Next.js instance in an empty project directory:
+
+```bash
+yarn create next-app my-project
+cd my-project
+```
+
+### Add TypeScript
+
+Keystone Next is built in TypeScript. Adding it as a `--dev` dependency will make Keystone implementation easier later on. Let‘s get that out of the way now:
+
+```bash
+yarn add --dev typescript @types/react
+```
+
+Rename the `.js` files in your `/pages` directory to `.tsx`.
+
+```text
+.
+└── pages
+   ├── _app.tsx
+   ├── api
+   │   └── hello.tsx
+   └── index.tsx
+```
+
+### Start your local server
+
+Run `yarn dev` at the root of your project. Next.js will generate a local server for you at [http://localhost:3000](http://localhost:3000).
+
+![A browser showing the Home page of the default Next.js app](/assets/tutorials/embedded-nextjs/localhost-home-1.png)
+
+## Add Keystone to your project
+
+Now that we have the Next.js starter with static files, let‘s embed Keystone into the app to blend file-based content with content you can edit using Keystone’s intuitive Admin UI.
+
+### Install dependencies
+
+Add the following Keystone dependencies to the project:
+
+```bash
+yarn add @keystone-next/keystone @keystone-next/admin-ui @keystone-next/fields
+```
+
+### Update .gitignore
+
+Add the `.keystone` directory to your `.gitignore` file. The contents of `.keystone` are dynamically generated at build time. You’ll never have to change them.
+
+```bash
+
+# in your .gitignore
+.keystone
+```
+
+### Add Keystone to Next.js config
+
+Add a `next.config.js` file to the project root with the following:
+
+```tsx
+
+// next.config.js
+
+const { withKeystone } = require('@keystone-next/keystone/next');
+
+module.exports = withKeystone();
+```
+
+This is where the magic happens – the `withKeystone` constant lets Next.js encapsulate Keystone in its script runtime, but Keystone still operates independently of the Next.js frontend.✨
+
+### Create a content List in Keystone config
+
+Now add a `keystone.ts` [configuration file](/apis/config) to the project root with a simple `Post` [list](/apis/schema) type. This will give us the ability to create and edit blog records in Keystone’s Admin UI:
```

Comment:
```suggestion
Now add a `keystone.ts` [configuration file](/apis/config) to the project root with a simple `Post` [list](/apis/schema). This will give us the ability to create and edit blog records in Keystone’s Admin UI:
```